### PR TITLE
Add s8 contexts in AGW

### LINKED ISSUE #3748: [Inbound Roaming][MME] Create s8_task
As discussed in the Inbound Roaming system architecture https://docs.google.com/document/d/1zJZpt5dCBjy54ODQ5DzbJZ2HufJgLxAIoVPyZPZL5UY/edit#heading=h.ysnq6v9qfm5k , we need to add SGW_S8 support in MME. Specifically,

- Add an SGW_S8 task which can communicate over GRPC with Roaming Relay service in Orc8r
- Enable SGW_S8 task to add S1<->S8 tunnel rules in OVS
- Add support for paging

### PR DESCRIPTION
**Pull request by rsarwad**
[lte][agw]: Add s8 contexts in AGW

## Summary
As part of this PR, created framework for handling S8 contexts.
Also added stateless framework for SGW-S8 task, but doesn't include conversion logic.

Contexts present sgw_context_manager.h are used in both  spgw and sgw_s8 tasks. So changed the location of file from tasks/sgw/sgw_context_manager.h to include/sgw_context_manager.h

## Test Plan
Executed s1ap sanity test suite

## Additional Information
Shall create new PR to handle S11 Create Session Request at SGW_S8 task

### CODE REVIEW

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

File: lte/gateway/c/oai/tasks/sgw_s8/sgw_s8_state_manager.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+constexpr char SGW_STATE_TABLE_NAME[] = "sgw_state";
+constexpr char SGW_TASK_NAME[]        = "SGW";
+}  // namespace
+
+namespace magma {
+namespace lte {
+
+/**
+ * SgwStateManager is a singleton (thread-safe, destruction guaranteed)
+ * that contains functions to maintain SGW and PGW state, allocating and
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/sgw_s8/sgw_s8_state_manager.h]:
```suggestion
 * that contains functions to maintain SGW state, allocating and
```

**Review by ssanadhya — APPROVED** [review]
Minor comment, rest looks good

**Pull request merged**